### PR TITLE
App refactoring

### PR DESCRIPTION
**Pull request by joaquimrocha**
The app was relying on the use of the node_modules from the frontend and was shipping the electron js twice by mistake.

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -2,33 +2,35 @@
...
   },
   "build": {
     "appId": "com.kinvolk.headlamp",
-    "extraMetadata": {
-      "main": "build/electron/main.js"
-    },
+    "beforeBuild": "./scripts/build-backend.js",
     "afterSign": "mac/scripts/notarize.js",
-    "asar": false,
+    "asar": true,
```
Comment (illume) [inline app/package.json:19]: We set asar false before to get source maps working. I'll check if sourcemaps are still working with the new setup.

File: app/package.json [hunk 2]
```diff
@@ -2,33 +2,35 @@
...
+    "artifactName": "${productName}-${version}-${os}-${arch}.${ext}",
```
Comment (illume) [inline app/package.json:20]: Do we really want these?

In the other PR https://github.com/kinvolk/headlamp/pull/214 I set this to just 
```json
    "artifactName": "${name}.${ext}",
```

Which makes eg. "Headlamp.app".

**Comment by illume** [comment]
I'm getting this error with `make app-linux`
```                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Error: Cannot find module '/home/rene/dev/kinvolk/headlamp/app/scripts/build-backend.js'
```

**Comment by illume** [comment]
Looking at the release notes for breaking changes in electron, doesn't look like there's anything they mentioned we're using.
- https://www.electronjs.org/releases/stable#release-notes-for-v1200
- https://www.electronjs.org/releases/stable?page=3#release-notes-for-v1100

**Comment by illume** [comment]
With the linux app, it doesn't seem to find the JavaScript.

File: app/package.json (hunk 1 above)
Comment (joaquimrocha) [reply app/package.json:20]: Ah right. I didn't remember this. Thanks.

File: app/package.json (hunk 2 above)
Comment (joaquimrocha) [reply app/package.json:21]: I do want it in this format as it matches the name we push when we release.

**Comment by joaquimrocha** [comment]
> With the linux app, it doesn't seem to find the JavaScript.

I have pushed the missing file.

File: app/package.json (hunk 1 above)
Comment (illume) [reply app/package.json:20]: The sourcemaps are working now. So we can keep the asar: true.

**Comment by illume** [comment]
I wiped the node_modules folders again and this time `make app-linux` is loading. I clicked about the app a bit and could not see any issues.

I'll try some other configurations.

**Comment by illume** [comment]
After quitting the app, the server process is still running for me. I had to manually kill it. Starting the app again after this means the app is nonfunctional.

**Review by illume — APPROVED** [review]
:+1:

**Comment by illume** [comment]
I tried the new commit with the linux builds (the tar.gz and AppImage) - it builds and runs.

**Comment by illume** [comment]
I see this warning when building:
```
  • description is missed in the package.json  appPackageFile=/home/rene/dev/kinvolk/headlamp/app/package.json
```

How about this? 

app/package.json
```json
  "description": "Headlamp is an easy-to-use and extensible Kubernetes web UI. Headlamp was created to be a Kubernetes web UI that has the traditional functionality of other web UIs/dashboards available (i.e. to list and view resources) as well as other features.",
```

**Comment by illume** [comment]
With a fresh clone, and `make app-linux` it has these warnings:
```
  • file source doesn't exist  from=/home/rene/dev/kinvolk/tmp/headlamp/frontend/build
  • file source doesn't exist  from=/home/rene/dev/kinvolk/tmp/headlamp/backend/server
  • default Electron icon is used  reason=application icon is not set
```

**Comment by joaquimrocha** [comment]
Right. That's because electron-builder is filtering out the node_modules inside each node_modules/DEPENDENCY..

**Pull request merged**